File: Samples/PasswordVault/js/js/scenario1-save.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    var PasswordVault = Windows.Security.Credentials.PasswordVault;
    var PasswordCredential = Windows.Security.Credentials.PasswordCredential;

    var inputResource;
    var inputUserName;
    var inputPassword;

    var page = WinJS.UI.Pages.define("/html/scenario1-save.html", {
        ready: function (element, options) {
            document.getElementById("SaveCredButton").addEventListener("click", saveCredential, false);
            inputResource = document.getElementById("InputResourceValue");
            inputUserName = document.getElementById("InputUserNameValue");
            inputPassword = document.getElementById("InputPasswordValue");
        }
    });

    function saveCredential() {
        //get input resource, username, password
        var resource = inputResource.value;
        var username = inputUserName.value;
        var password = inputPassword.value;

        if (resource === "" || username === "" || password === "") {
            WinJS.log && WinJS.log("All fields are required when saving a credential.", "sample", "error");
            return;
        }

        // Add a credential to PasswordVault with provided resource, user name, and password.
        // Replaces any existing credential with the same resource and user name.
        var vault = new PasswordVault();
        var cred = new PasswordCredential(resource, username, password);
        vault.add(cred);
        WinJS.log && WinJS.log("Credential saved successfully. Resource: " + cred.resource + ", UserName: " + cred.userName + ", Password: " + cred.password, "sample", "status");
    }
})();

File: Samples/PasswordVault/cppwinrt/Scenario1_Save.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario1_Save.h"
#include "Scenario1_Save.g.cpp"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::UI::Xaml;
using namespace Windows::Security::Credentials;

namespace winrt::SDKTemplate::implementation
{
    Scenario1_Save::Scenario1_Save()
    {
        InitializeComponent();
    }

    void Scenario1_Save::SaveCredential_Click(IInspectable const&, IInspectable const&)
    {
        auto resource = InputResourceValue().Text();
        auto userName = InputUserNameValue().Text();
        auto password = InputPasswordValue().Password();

        if (resource.empty() || userName.empty() || password.empty())
        {
            rootPage.NotifyUser(L"All fields are required when saving a credential.", NotifyType::ErrorMessage);
        }
        else
        {
            // Add a credential to PasswordVault with provided resource, user name, and password.
            // Replaces any existing credential with the same resource and user name.
            PasswordVault vault;
            PasswordCredential cred{ resource, userName, password };
            vault.Add(cred);

            rootPage.NotifyUser(L"Credential saved successfully. Resource: " + cred.Resource() + L" Username: " + cred.UserName() + L" Password: " + cred.Password(), NotifyType::StatusMessage);
        }
    }
}

